py/compile.c

```diff
@@ -1633,6 +1636,9 @@ STATIC void compile_try_except(compiler_t *comp, mp_parse_node_t pn_body, int n_
             EMIT_ARG(pop_jump_if, false, end_finally_label);
         }
 
+#if MICROPY_PY_SYS_SETTRACE
+        EMIT_ARG(set_source_line, pns_except->source_line);
```

Comment:
I see how this is useful for settrace.  For the following code:
```python
try:
    raise 1
except Exceptio as er:
    pass
```
It adds the line number info for the "except Exceptio as er:" line, which in this case gives better error reporting when the lookup of `Exceptio` fails.

Actually it'd be better if it were moved up to line 1614.